smarts/sstudio/types.py

```diff
@@ -393,6 +394,57 @@ def __eq__(self, other):
         return self.__class__ == other.__class__ and hash(self) == hash(other)
 
 
+@dataclass(frozen=True)
+class SingleVehicle:
+    """A single vehicle that can be interchanged with flow."""
+
+    route: Union[RandomRoute, Route]
+    """The route that the vehicle will use."""
+
+    actors: Dict[TrafficActor, float] = field(default_factory=dict)
+    """An actor to weight mapping associated as\\: { actor: weight }
+    actor:
+        The traffic actors that are provided.
+    weight:
+        The chance of this actor appearing as a ratio over total weight.
+    """
+
+    depart: float = 0
+
+    @property
+    def id(self) -> str:
+        return "single-vehicle-{}-{}-".format(
+            self.route.id,
+            str(_pickle_hash(sorted(self.actors.items(), key=lambda a: a[0].name))),
+        )
+
+    def __hash__(self):
+        # Custom hash since self.actors is not hashable, here we first convert to a
+        # frozenset.
+        return _pickle_hash((self.route, self.rate, frozenset(self.actors.items())))
+
+    def __eq__(self, other):
+        return self.__class__ == other.__class__ and hash(self) == hash(other)
+
+
+@dataclass
+class Rerouter:
+    """A route changing pad that will change the end route of an actor."""
+
+    begin: Tuple[str, int, Any]
+    """The rerouter location."""
```

Comment:
I'm assuming this is `(road, lane_index, offset)` right?  (If so, why are we using `Any` for offset instead of `float`?)  I think you should indicate the semantics of the tuple in the comment.